src/keycodes.rs

```diff
@@ -1,111 +1,375 @@
+        Keycode::F6 => LogicalKey::Named(NamedKey::F6),
+        Keycode::F7 => LogicalKey::Named(NamedKey::F7),
+        Keycode::F8 => LogicalKey::Named(NamedKey::F8),
+        Keycode::F9 => LogicalKey::Named(NamedKey::F9),
+        Keycode::F10 => LogicalKey::Named(NamedKey::F10),
+        Keycode::F11 => LogicalKey::Named(NamedKey::F11),
+        Keycode::F12 => LogicalKey::Named(NamedKey::F12),
+        Keycode::NumLock => LogicalKey::Named(NamedKey::NumLock),
+        Keycode::Numpad0 => LogicalKey::Character('0'),
+        Keycode::Numpad1 => LogicalKey::Character('1'),
+        Keycode::Numpad2 => LogicalKey::Character('2'),
+        Keycode::Numpad3 => LogicalKey::Character('3'),
+        Keycode::Numpad4 => LogicalKey::Character('4'),
+        Keycode::Numpad5 => LogicalKey::Character('5'),
+        Keycode::Numpad6 => LogicalKey::Character('6'),
+        Keycode::Numpad7 => LogicalKey::Character('7'),
+        Keycode::Numpad8 => LogicalKey::Character('8'),
+        Keycode::Numpad9 => LogicalKey::Character('9'),
+        Keycode::NumpadDivide => LogicalKey::Character('/'),
+        Keycode::NumpadMultiply => LogicalKey::Character('*'),
+        Keycode::NumpadSubtract => LogicalKey::Character('-'),
+        Keycode::NumpadAdd => LogicalKey::Character('+'),
+        Keycode::NumpadDot => LogicalKey::Character('.'),
+        Keycode::NumpadComma => LogicalKey::Character(','),
+        Keycode::NumpadEnter => LogicalKey::Named(NamedKey::Enter),
+        Keycode::NumpadEquals => LogicalKey::Character('='),
         _ => return None,
+    };
+
+    let key_location = match android.key_code() {
+        Keycode::AltLeft | Keycode::ShiftLeft | Keycode::CtrlLeft | Keycode::MetaLeft => {
+            KeyLocation::Left
+        }
+
+        Keycode::AltRight | Keycode::ShiftRight | Keycode::CtrlRight | Keycode::MetaRight => {
+            KeyLocation::Right
+        }
+
+        Keycode::NumLock
+        | Keycode::Numpad0
+        | Keycode::Numpad1
+        | Keycode::Numpad2
+        | Keycode::Numpad3
+        | Keycode::Numpad4
+        | Keycode::Numpad5
+        | Keycode::Numpad6
+        | Keycode::Numpad7
+        | Keycode::Numpad8
+        | Keycode::Numpad9
+        | Keycode::NumpadDivide
+        | Keycode::NumpadMultiply
+        | Keycode::NumpadSubtract
+        | Keycode::NumpadAdd
+        | Keycode::NumpadDot
+        | Keycode::NumpadComma
+        | Keycode::NumpadEnter
+        | Keycode::NumpadEquals
+        | Keycode::NumpadLeftParen
+        | Keycode::NumpadRightParen => KeyLocation::Numpad,
+
+        _ => KeyLocation::Standard,
+    };
+
+    Some(KeyDescriptor {
+        physical_key,
+        logical_key,
+        key_location,
     })
 }
+
+pub fn key_tag_to_key_descriptor(tag: &str) -> Option<KeyDescriptor> {
+    match tag {
+        "A" => Some(KeyDescriptor {
+            physical_key: PhysicalKey::KeyA,
+            logical_key: LogicalKey::Character('a'),
```

Comment:
Does this virtual keyboard have support for uppercase letters? If yes then the logical key should take that into account